src/cargo/util/command_prelude.rs

```diff
@@ -466,6 +473,15 @@ pub trait ArgMatchesExt {
                 .fail_if_stable_opt("--unit-graph", 8002)?;
         }
 
+        if self._is_present("future-incompat-report") {
+            if !config.cli_unstable().future_incompat_report {
+                anyhow::bail!(
+                    "Usage of `--future-incompat-report` requires `-Z future-incompat-report`"
+                )
+            }
+            build_config.future_incompat_report = true;
+        }
```

Comment:
We already have `-Z future-incompat-report` - using the same code as the previous opts will require `-Z unstable-opts` as well, which seems redundant.

Reply:
I think it would be better to pattern this after the other options because `--future-incompat-report` is an unstable option and the unstable error message is more consistent with the above pattern.

Reply:
I think there's been some confusion about exactly which flags do what. This is my proposal:

1. The output of a plain `cargo build` on Nightly is unchanged - that is, we do *not* print any kind of message suggesting that the user display the full future-incompat-report, even if one was generated.
2. To display this message, the user needs to run `cargo build -Z future-incompat-report`, which enables the logic for generating the report, and notifying the user about it.
3. Since we already have a `-Z` flag, we can use it to gate the `--future-incompat-report` flag for displaying the report - e.g. `cargo build -Z future-incompat-report --future-incompat-report`.
4. Optionally, `cargo describe-future-incompat` can require `-Z future-incompat-report`, or can be nightly only.
If I understand your proposal correctly, we would need to either:

1. Use the flag as `cargo build -Z future-incompat-report -Z unstable-opts --future-incompat-report`, which seems incredibly verbose.
2. Use the flag as `cargo build -Z unstable-opts --future-incompat` report. If we allow `cargo describe-future-incompat` on nightly without any flags, then the only usage of `-Z future-incompat-report` is for displaying the notice (but not the full report) via `cargo build -Z future-incompat-report`.

My motivation here was to avoid changing the behavior of any existing commands without explicit opt-in. If you wanted, we could simplify things by having `cargo build` print the future-incompat notice on Nightly, without requiring any flags.

Reply:
Yes I understand what this PR is doing, and I'm saying that this should be:

```
cargo build -Z unstable-opts --future-incompat-report
```

Cargo can internally recognize that a report is being generated and implicitly enable  `-Z future-incompat-report`. There's not really any reason to additionally require `-Z future-incompat-report`. I'm fine requiring it as well though if you'd prefer. Commands should indeed remain as they are unless new flags are passed.

Reply:
I've changed the code to require voth `-Z unstable-opts` and `-Z future-incompat-report`.